### PR TITLE
Prime: Add power beam only doors to DLR

### PR DESCRIPTION
**Pull request by JeffGainsNGames**
expose the power beam only doors to Randovania. People shuffle power beam, think this is a good addition.

### CODE REVIEW

File: randovania/games/prime1/logic_database/header.txt [hunk 1]
```diff
@@ -82,6 +82,13 @@ Dock Weaknesses
       No lock
 
 
+  * Power Door
```
Comment (Miepee) [inline randovania/games/prime1/logic_database/header.txt:85]: IMO `Power Beam Door` sounds nicer

Comment (JeffGainsNGames) [reply randovania/games/prime1/logic_database/header.txt:85]: yes, but it doesn't fit with all the others `wave door, plasma door, ice door`

Comment (Miepee) [reply randovania/games/prime1/logic_database/header.txt:85]: The existing ones should be renamed too IMO

**Review by duncathan — CHANGES_REQUESTED** [review]
I've mentioned my position on this before: I'm perfectly happy with it so long as they cannot be included unless power beam is shuffled. Same goes for e.g. charge doors in echoes. Until such a system is implemented I'm against it, though

**Comment by duncathan** [comment]
One possible solution might be to exclude weaknesses that can be opened with the starting equipment. If variety is a concern, you could pick from those weaknesses any time a blue door would be placed.

**Comment by JeffGainsNGames** [comment]
Is it really such a bad thing to have it available always?
What if someone wants to play with vanilla item pool but shuffle the doors anyways?
Is there also no responsibility of the player to ensure they pick doors that make sense with there item pool?

**Comment by Miepee** [comment]
> I'm perfectly happy with it so long as they cannot be included unless power beam is shuffled

With the way RDV is laid out, I very much dislike options changing based on what items are in the pool. IMO, if someone wants to shuffle power beam-only doors even tho they start with it, then they should be allowed to do so.

**Comment by duncathan** [comment]
> > I'm perfectly happy with it so long as they cannot be included unless power beam is shuffled
> 
> With the way RDV is laid out, I very much dislike options changing based on what items are in the pool. IMO, if someone wants to shuffle power beam-only doors even tho they start with it, then they should be allowed to do so.

In terms of UI, you mean? Do you have any thoughts on my suggested implementation?

**Comment by duncathan** [comment]
> Is it really such a bad thing to have it available always? What if someone wants to play with vanilla item pool but shuffle the doors anyways? Is there also no responsibility of the player to ensure they pick doors that make sense with their item pool?

I pretty firmly believe that we should avoid giving players the ability to pick settings that aren't appropriate. I see it as a failing of some of our design that it's so easy to do atm. As for your first point, did you have any thoughts about my suggested implementation?

**Comment by Miepee** [comment]
> In terms of UI, you mean? 

Not just in terms of UI, but also in general. It feels very unintuitive that Door Lock related features are linked to what is in the Pool. I personally would see having that much more as a design flaw than player being able to do whatever unappropriate settings they want to do.

> Do you have any thoughts on my suggested implementation?

How would that play out with Door Type Rando?

File: randovania/games/prime1/logic_database/header.json [hunk 2]
```diff
@@ -1439,6 +1439,16 @@
                         },
                         "lock": null
                     },
+                    "Power Door": {
```
Comment (toasterparty) [inline randovania/games/prime1/logic_database/header.json:1442]: Power Door would imply that canonically, Power Beam is the *minimum* requirement to open such a door. Future maintainers of this file may get this confused with vanilla blue doors.

Also, Power Door could be confused with the "Powered" doors in crashed frigate etc.

Please use `Power Beam Only` for any and all code/UI which references this new door

**Review by toasterparty — CHANGES_REQUESTED** [review]

**Review by toasterparty — APPROVED** [review]

**Pull request merged**